src/components/Sidebar/BranchTransition.tsx

```diff
@@ -1,21 +1,119 @@
-import React, { FC, memo } from 'react';
+import React, { FC, memo, useState, useCallback, ChangeEvent, useEffect, useRef } from 'react';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faPlus, faTools } from '@fortawesome/free-solid-svg-icons';
-
+import DropDown from 'components/elements/DropDown';
+import { TextField, FormControl } from '@material-ui/core';
+import { setTransitionCriteria, setTransitionCriteriaDisplay } from 'utils/builder';
 import { SidebarHeader, SidebarButton } from '.';
 import { Pathway, Transition } from 'pathways-model';
 import useStyles from './styles';
 
 interface BranchTransitionProps {
   pathway: Pathway;
+  currentNodeKey: string;
   transition: Transition;
   updatePathway: (pathway: Pathway) => void;
 }
 
-const BranchTransition: FC<BranchTransitionProps> = ({ pathway, transition, updatePathway }) => {
+const criteriaOptions = [
```

Comment:
Now that you rebased and have the criteria provider to work with, I think it's just a couple lines to integrate it here.

```jsx
import { useCriteriaContext } from 'components/CriteriaProvider';

  const { criteria } = useCriteriaContext();
  const criteriaOptions = criteria.map(c => ({ value: c.label, label: c.label })); // or something else along these lines
```